[Dynamic Dashboard] Error view for when WCAdmin is disabled

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

Part of #11544 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
This PR adds the following changes:
1. Introduces a common error view to use when WCAdmin is not available and the card's API depends on it.
2. Updates the Performance and Top Performers cards to use the new component.
3. Updates the Coupons card to use it too.

### Testing instructions
1. Disable WCAdmin (in a non-atomic website, WooCommerce -> Settings -> Advanced -> Features -> Enable WooCommerce Analytics)
2. Open the app.
3. Test Performance, Top Performers and Coupons cards.
4. Confirm all show the correct error view.

### Images/gif
<img src="https://github.com/woocommerce/woocommerce-android/assets/1657201/74229c32-301f-4c82-b93f-5130fbd3bd67" width=360 />

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Introduce a common component for the error view when WCAdmin is disabled

hichamboushaba committed:
Update the performance card to use the new component

hichamboushaba committed:
Use a specific error in the top performers card when WCAdmin is disabled

hichamboushaba committed:
Update the coupons card to use the correct error view

hichamboushaba committed:
Fix the logic of calculating `isError`

dangermattic commented:
<!--
  1 Error: This PR is tagged with <code>s...
  1 Warning: Class <code>ErrorType</code> i...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>status: do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ErrorType</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes commented on WooCommerce/src/main/res/values/strings.xml:
```diff
@@ -387,14 +387,13 @@
     <string name="dashboard_action_view_all_orders">View all orders</string>
 
     <string name="my_store_stats_plugin_inactive_title">We can\'t display your\n store\'s analytics</string>
-    <string name="my_store_stats_plugin_inactive_description">Make sure you are running the latest version of WooCommerce on your site and that you have WooCommerce Admin activated.</string>
-    <string name="my_store_stats_plugin_inactive_contact_us">Still need help? Contact us</string>
```
Hmm, curious why you removed this 2 tokens from the rest of the languages if neither their id or its value has changed? What am I missing?

hichamboushaba replied:
I did just to be safe that lint won't be unhappy about the change, I'm not sure if it's needed, but I thought it's better to be safe than causing some delay during the code freeze because of it.

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/WCAdminNotEnabledErrorView.kt:
```diff
@@ -0,0 +1,67 @@
+import com.woocommerce.android.ui.compose.theme.WooThemeWithBackground
+
+@Composable
+fun WCAdminNotAvailableErrorView(
```
May I ask why are we referrring to this error as `WCAdminNotAvailable` ? Considering the setting we are disabling is `WooCommerce analytics` to trigger this error type, why are all references in the code as well as the error message for the user talking about `WooCommerce admin activated`?

hichamboushaba replied:
That's actually a very good point, and after doing some research, I found out that my information on the subject is outdated, here is the up-to-date information:
- Analytics are part of WCAdmin, and disabling WCAdmin disables them.
- Analytics can be disabled indepently, which is what the option in the settings does.

This is relatively [new](https://developer.woocommerce.com/2021/05/18/request-for-comments-removing-the-filter-to-turn-off-woocommerce-admin/), and that's why in the app we had the habitude of calling anything about the Analytics `WooCommerce Admin`, but I think we need to update the wording now for more clarity, I'll do it, and I'll update the name of the errors as well.

hichamboushaba replied:
Done in 6d0c7570104ab69d8ccb05423ccbe16953013c86

JorgeMucientes commented:
Great job @hichamboushaba 👏🏼 The error is shown as expected and the code looks good. Ready for merging. 

I just left a couple of questions for my learning. But nothing blocking.

hichamboushaba committed:
Rename all occurences of WCAdmin with WCAnalytics

The option in Core is called WooCommerce Analytics, it's still part of WooCommerce Admin, but referring to the specific feature will be easier for the users to find the correct option.
